src/libCli/ConnectionManager.hpp

```diff
@@ -0,0 +1,163 @@
+// Copyright 2019 IBM Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <third_party/gRPC_utils/proto_reflection_descriptor_database.h>
+
+namespace cli
+{
+    /// List of connection infomation
+    typedef struct ConnList
+    {
+       std::shared_ptr<grpc::Channel> channel = nullptr;
+       std::shared_ptr<grpc::ProtoReflectionDescriptorDatabase> descDb = nullptr;
+       std::shared_ptr<grpc::protobuf::DescriptorPool> descPool = nullptr;
+
+    } ConnList;
+
+    /// Class to manage and resuse connection information, singleton pattern.
+    class ConnectionManager
+    {
+        public:
+            ConnectionManager(const ConnectionManager & ) = delete;
+            ConnectionManager& operator=(const ConnectionManager & ) = delete;
+        private:
+            ConnectionManager(){}
+            ~ConnectionManager(){}
+
+        public:
+            /// Only use a single connection instance
+            static ConnectionManager & getInstance()
+            {
+                static ConnectionManager connectionManager;
+                return connectionManager;
+            }
+
+            bool findChannelByAddress(std::string f_address)
+            {
+                if(connections.find(f_address) != connections.end())
+                {
+                    if(connections[f_address].channel != nullptr)
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+            bool findDescDbByAddress(std::string f_address)
+            {
+                if(connections.find(f_address) != connections.end())
+                {
+                    if(connections[f_address].descDb != nullptr)
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+            bool findDescPoolByAddress(std::string f_address)
+            {
+                if(connections.find(f_address) != connections.end())
+                {
+                    if(connections[f_address].descPool != nullptr)
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+
```

Comment:
Maybe it is a good idea to make all methods above this point private, as currently they are only used internally